index.rst

```diff
@@ -46,14 +46,314 @@
+These failures would mostly manifest once a sufficiently large number of more intensive components (e.g. MTM1M3 and MTMount) where brought up in the system.
+Initially, they where observed mainly by users trying to run control sequences from Jupyter notebooks in nublado, though it was soon realized that the problem affected all other components in the system (salkafka producers, CSCs running SAL and SalObj).
+
+Following, are some details of the history and experience accumulated over time with DDS, results of the investigation conducted at NCSA test stand about this new failure mode and the proposed solution.
+
+.. note::
+
+   These tests occurred in the week of July 20-24, 2020.
+
+.. _Overview:
+
+Overview
+========
+
+The test started by using the OpenSplice standard deployment, which employ the DDS :ref:`single process mode <fig-single-process>`.
+This operation mode has been used with the Vera Rubin Observatory system from the beginning and, despite some occasional issues, worked relatively well over time.
+In fact, most of the time we encountered issues with the DDS communication we could relate them to one of the following issues:
+
+  * Wrong DDS partition.
+
+    The DDS partition is controlled by SAL and SalObj by the ``LSST_DDS_DOMAIN`` system environment variable.
+    It was common on some occasions that users would set different values for that variable, which causes DDS to filter the messages and communication would not work.
+    The solution is to make sure everyone on the system has the same values for the environment variable.
+
+  * Systems running incompatible versions of xml interface.
+
+    The xml interface determines the DDS topics data structure.
+    If topics with the same identity have different data structures the last system to try to connect to the DDS network will fail.
+    This created several issues in the past as DDS cache would prevent users from upgrading the topic data structure.
+    A mitigation to this problem was introduced in recent versions of SAL where topics identity was attached to a hash code based on its content.
+    This allowed users to upgrade topics without colliding with previous cached versions but also meant that incompatible versions silently fail to communicate.
+
+  * Systems running with incompatible versions of SAL/SalObj.
+
+    SAL and SalObj are responsible for managing topics Quality of Service (QoS) parameters.
+    The system defines 4 different types of QoS depending on the "type" of topic: Command, Event, Telemetry and Acknowledgement.
+    Topics with different QoS may sometimes be filtered out by the DDS reader so, an incompatible version of SAL/SalObj may not be able to exchange some messages.
+    This was improved considerably in more recent versions of SAL/SalObj and we have been able to keep breakages to a minimum, especially by rigidly managing deployments.
+
+  * Wrong network attached.
```

Comment:
For all of these situations, is it worth adding one sentence on how you diagnose that this is the issue if it occurs? 
It might be useful to get a sense just how bad it is to diagnose these, then solve them.... although maybe there isn't space (or this isn't the place?)